packages/core/src/modules/credentials/protocol/v2/CredentialMessageBuilder.ts

```diff
@@ -243,8 +243,10 @@ export class CredentialMessageBuilder {
     // Construct v2 offer message
     const message = new V2OfferCredentialMessage(messageProps)
 
+    const id = connection ? connection.id : undefined
+
     const recordProps: CredentialExchangeRecordProps = {
-      connectionId,
+      connectionId: id,
```

Comment:
I think we use optional chaining here

```suggestion
      connectionId: connection?.id,
```